.github/workflows/ci.yml

```diff
@@ -0,0 +1,33 @@
+on:
+  push:
+    branches: [ staging, trying, master ]
+  pull_request:
+
+name: CI
+
+jobs:
+  ci-linux:
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        # All generated code should be running on stable now
+        rust: [stable]
+
+        include:
+          # Test MSRV
+          - rust: 1.38.0
+
+          # Test nightly but don't fail
+          - rust: nightly
+            experimental: true
+    steps:
+      - uses: actions/checkout@v2
+      - uses: actions-rs/toolchain@v1
+        with:
+          profile: minimal
+          toolchain: ${{ matrix.rust }}
+          override: true
+      - name: Run tests
+        run: cargo test --all
+
+# FIXME: test on macOS and Windows
```

Comment:
Unlike the Travis config this doesn't have a cache enabled, but since cortex-m has few and simple dependencies maybe there's not much advantage.

Reply:
Yeah, I want to avoid dealing with caching since it can be quite unreliable

Reply:
Fine with me.

Reply:
CI time went down to ~2 minutes with this. Looks like we don't need caching for now :smile: 

(maybe this will change once we import cortex-m-rt, but I doubt it)